pyupgrade.py

```diff
@@ -852,26 +852,56 @@ def _fix_super(contents_text):
     return tokens_to_src(tokens)
 
 
-class FindNewStyleClasses(ast.NodeVisitor):
+class SixVisitor(ast.NodeVisitor):
+    def __init__(self):
+        self.six_from_imports = set()
+
+    def visit_ImportFrom(self, node):
+        if node.module == 'six':
+            for name in node.names:
+                if not name.asname:
+                    self.six_from_imports.add(name.name)
+        self.generic_visit(node)
+
+
+class FindClasses(SixVisitor):
     Base = collections.namedtuple('Base', ('node', 'index'))
 
     def __init__(self):
+        super(FindClasses, self).__init__()
         self.found = {}
 
     def visit_ClassDef(self, node):
         for i, base in enumerate(node.bases):
-            if isinstance(base, ast.Name) and base.id == 'object':
+            if (
+                (
+                    isinstance(base, ast.Name) and
+                    (
+                        base.id == 'object' or
+                        (
+                            base.id == 'Iterator' and
+                            'Iterator' in self.six_from_imports
+                        )
```

Comment:
let's keep these cases separate just so it's a little easier to see what's going on -- I realize you golfed down the `isinstance(..., ast.Name)` but I think it's less readable now.

Also let's do `base.id in self.six_from_imports` for the last condition

Reply:
sure